### DISCUSSION TITLE
How to get the complete receiving information

### DISCUSSION THREAD

**Discussion by Kepler-Elliott** [amzn/selling-partner-api-models#4334] [Q&A]
When I get the delivery address, the returned data will have some parameter data is not returned, such as the name is returned but is empty, the street field is not returned

**Comment by jawheral** [comment]
Hi @Kepler-Elliott,

Could you please provide more information:
- Which API or report are you using? 
- What use case are you trying to achieve?
- Which data do you need?

Best,
Jawher
Selling Partner Developer Services

**Reply by Kepler-Elliott** [reply]
Hi @jawheral 
What I want to get is the address of the order. I use the api /orders/v0/orders/{orderId}/address. I've got The rdt but the data returned by putting it in the request header is {"errors":[{"code":"Unauthorized","details":"The access token you provided is revoked, malformed or invalid.","message":"Access to requested resource is denied."}]}

**Reply by jawheral** [reply]
@Kepler-Elliott
You should be able to call `/orders/v0/orders/{orderId}/address` without an RDT.
Did you make sure that your dev profile and application have the [Direct to Consumer Shipping](https://developer-docs.amazon.com/sp-api/docs/direct-to-consumer-shipping-restricted-role) role?

Could you elaborate more on your use case? Why do you need the order address?

Best,
Jawher
Selling Partner Developer Services

**Comment by Kepler-Elliott** [comment]
@jawheral 
So I didn't get it because I didn't have the Direct to Consumer Shipping role? How do I check if I have this permission

**Reply by jawheral** [reply]
@Kepler-Elliott

You can check the roles by navigating to:
1 - Developer Central > View profile -> roles
2 - Edit App -> Role

Sellers who authorized the application before the roles were added need to reauthorize the application again so the application can call the role-bound resources on their behalf.

Best,
Jawher
Selling Partner Developer Services

**Reply by Kepler-Elliott** [reply]
@jawheral 
I want to know our authorized application id where I can get, I saw this interface createRestrictedDataToken need targetApplication this parameter

**Reply by jawheral** [reply]
Hi @Kepler-Elliott,

On Developer Central, the app id is the one that comes under the application name.

<img width="336" alt="Screenshot 2024-11-21 at 10 53 43" src="https://github.com/user-attachments/assets/36c0e54f-021c-4040-8066-ebe04fcd8a73">

That said, why do you need to delegate access to another application id using RDTs? 
As mentioned before, you don't require an RDT to retrieve the `OrderAddress` using the `getOrderAddress` operation. Making sure your developer profile, application have the right restricted role and a fresh authorization (including the role) would be sufficient to access the data.


Best,
Jawher
Selling Partner Developer Services

**Reply by Kepler-Elliott** [reply]
Hi @jawheral Because I see that some fields of the shippingaddress seem to be restricted. I see the example of the interface that says to use the rdt as the request header to access, as you said, if I directly request, I can get the data but only get the unrestricted data, but I also want to get the detailed address information

**Reply by Kepler-Elliott** [reply]
This is the result I returned, this result whether or not mean that I do not have permission @
![1](https://github.com/user-attachments/assets/01e0df63-ac2c-482d-aa31-7887d9134f21)

**Reply by Kepler-Elliott** [reply]
I also want to know what the RestrictedResource parameter dataElements should be, is it the restrictedresource field that is currently requested

**Comment by Kepler-Elliott** [comment]
The first line is the parameter that I passed, and the last line is the request that I returned. Why is that
![2](https://github.com/user-attachments/assets/3a7a8d94-16b1-4210-adf6-0687de101a66)

**Reply by Kepler-Elliott** [reply]
@jawheral 
I think I got it. Thank you

**Comment by Kepler-Elliott** [comment]
@jawheral 
Hello, when I got the rdt, I filled path with /reports/2021-06-30/documents/"+reportDocumentId, But what he returns is this {"errors":[{"code":"InvalidInput","details":"","message":"Invalid or unsupported document ID found"}]} Why

**Reply by weilinggu** [reply]
Hi @Kepler-Elliott,
Are you following the tutorial [here](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-tutorial-request-a-report) on the documentation site on requesting a report?
If you still have problem after following the tutorial, it might be worth [opening a support case](https://developer.amazonservices.com/support) following the [guidance](https://developer.amazonservices.com/support-guidelines) here so developer support can help you troubleshoot the issue.
Thanks,
Weiling
Amazon Selling Partner Services
